app/main.py

```diff
@@ -1,34 +1,88 @@
+from typing import Optional
+
+ROWS = 10
+COLUMNS = 10
+
+
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(
+            self,
+            row: int,
+            column: int,
+            is_alive: bool = True
+    ) -> None:
+        self.row = row
+        self.column = column
+        self.coordinates = (row, column)
+        self.is_alive = is_alive
+
+    def __repr__(self) -> str:
+        return f"Deck({self.row}, {self.column}) - alive: {self.is_alive} |"
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(
+            self,
+            start: tuple[int, int],
+            end: tuple[int, int],
+            is_drowned: bool = False
+    ) -> None:
+        if start == end:
+            self.decks = [Deck(*start)]
+        elif start[0] == end[0]:
+            self.decks = [
+                Deck(start[0], column)
+                for column in range(start[1], end[1] + 1)
+            ]
+        else:
+            self.decks = [
+                Deck(row, start[1])
+                for row in range(start[0], end[0] + 1)]
```

Comment:
This part of code looks really heavy for constructor. I would prefer to move this to some method